akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala

```diff
@@ -6,42 +6,45 @@ package akka.stream
 /**
  * Represents a strategy that decides how to deal with a buffer that is full but is about to receive a new element.
  */
-sealed abstract class OverflowStrategy
+sealed abstract class OverflowStrategy extends Serializable
+sealed trait DelayOverflowStrategy extends Serializable
 
-object OverflowStrategy {
+private[akka] trait BaseOverflowStrategy {
 
   /**
    * INTERNAL API
    */
-  private[akka] final case object DropHead extends OverflowStrategy
```

Comment:
@agolubev I asked the question once and akka team said that they mark it final for internal thing,